src/server/game/Entities/AreaTrigger/AreaTrigger.cpp

```diff
@@ -212,11 +213,60 @@ AreaTrigger* AreaTrigger::CreateAreaTrigger(uint32 spellMiscId, Unit* caster, Un
     return at;
 }
 
+bool AreaTrigger::CreateServer(Map* map, AreaTriggerTemplate const* areaTriggerTemplate, AreaTriggerServerPosition const& position)
```

Comment:
Maybe an idea to merge this functionality in `AreaTrigger::Create`? Add an extra bool as argument and then use the position passed into that function as server position.

Reply:
This is the current argument list of the `AreaTrigger::Create` function:
`uint32 spellMiscId, Unit* caster, Unit* target, SpellInfo const* spell, Position const& pos, int32 duration, SpellCastVisual spellVisual, ObjectGuid const& castId, AuraEffect const* aurEff`

The only one that can be provided is the position, and even with just that and the bool info would be lacking since it's getting a `AreaTriggerMiscTemplate` from the spell, and using that to get the script Id, which is irrelevvant with the server side scripts.

Reply:
Hmmm, okay, we'll see what Shauren or any other contributer has to say heh :D